File: ea-fel/src/de/proskor/fel/stub/ArchitecturalViewImpl.java
package de.proskor.fel.stub;

import java.util.ArrayList;
import java.util.List;

import de.proskor.fel.container.EventTypeContainer;
import de.proskor.fel.view.ArchitecturalView;

public class ArchitecturalViewImpl extends ViewImpl implements ArchitecturalView {
	private List<EventTypeContainer> components = new ArrayList<EventTypeContainer>();

	@Override
	public List<EventTypeContainer> getEventTypeContainers() {
		return this.components;
	}
}

File: fel-adapter/src/de/proskor/fel/impl/ArchitecturalViewImpl.scala
package de.proskor.fel.impl

import java.util.{List => JavaList}
import scala.collection.JavaConversions._
import de.proskor.automation.Diagram
import de.proskor.fel.container.EventTypeContainer
import de.proskor.fel.view.ArchitecturalView

class ArchitecturalViewImpl(diagram: Diagram) extends ViewImpl(diagram) with ArchitecturalView {
  override def getEventTypeContainers: JavaList[EventTypeContainer] = {
    val list = for {
      node <- diagram.nodes
      element = node.element
      if element.stereotype == "EventTypeContainer"
    } yield new EventTypeContainerImpl(element)
    list.toList
  }
}